nixos/modules/services/misc/gitit.nix

```diff
@@ -0,0 +1,598 @@
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 5001;
+        description = "Port on which the web server will run";
+      };
+
+      wikiTitle = mkOption {
+        type = types.str;
+        default = "Gitit!";
+        description = "The wiki title";
+      };
+
+      repositoryType = mkOption {
+        type = types.enum ["git" "darcs" "mercurial"];
+        default = "git";
+        description = "Specifies the type of repository used for wiki content";
+      };
+
+      repositoryPath = mkOption {
+        type = types.path;
+        default = "/var/lib/gitit/wiki";
+        description = ''
+          Specifies the path of the repository directory. If it does not
+          exist, gitit will create it on startup.
+        '';
+      };
+
+      requireAuthentication = mkOption {
+        type = types.enum [ "none" "modify" "read" ];
+        default = "modify";
+        description = ''
+          If 'none', login is never required, and pages can be edited anonymously.
+          If 'modify', login is required to modify the wiki (edit, add, delete
+          pages, upload files).
+          If 'read', login is required to see any wiki pages.
+        '';
+      };
+
+      authenticationMethod = mkOption {
+        type = types.enum [ "form" "http" "generic"];
+        default = "form";
+        description = ''
+          'form' means that users will be logged in and registered
+          using forms in the gitit web interface.  'http' means
+          that gitit will assume that HTTP authentication is in
+          place and take the logged in username from the "Authorization"
+          field of the HTTP request header (in addition,
+          the login/logout and registration links will be
+          suppressed).  'generic' means that gitit will assume that
+          some form of authentication is in place that directly
+          sets REMOTE_USER to the name of the authenticated user
+          (e.g. mod_auth_cas on apache).  'rpx' means that gitit
+          will attempt to log in through https://rpxnow.com.
+          This requires that 'rpx-domain', 'rpx-key', and 'base-url'
+          be set below, and that 'curl' be in the system path.
+        '';
+      };
+
+      userFile = mkOption {
+        type = types.path;
+        default = "gitit-users";
+        description = ''
+          Specifies the path of the file containing user login information.
+          If it does not exist, gitit will create it (with an empty user list).
+          This file is not used if 'http' is selected for authentication-method.
+        '';
+      };
+
+      sessionTimeout = mkOption {
+        type = types.int;
+        default = 60;
+        description = ''
+          Number of minutes of inactivity before a session expires.
+        '';
+      };
+
+      staticDir = mkOption {
+        type = types.path;
+        default = "static";
```

Comment:
I thought types.path meant absolute paths...? Anyway, if a relative path is accepted, maybe the description field should say what it is relative to (repositoryPath?).